src/rgw/rgw_op.cc

```diff
@@ -5036,12 +5036,14 @@ void RGWPutACLs::execute()
       return;
 
     new_data = strdup(ss.str().c_str());
-    free(data);
-    data = new_data;
     len = ss.str().size();
+    if (data) {
+      data.reset(nullptr);
+    }
+    data.reset(new_data);
```

Comment:
new_data was allocated by strdup() (ie malloc), and unique_ptr<char[]> will try to free it with delete[]

Reply:
oops.